sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -929,7 +929,7 @@ class CodegenContext {
     // be extremely expensive in certain cases, such as deeply-nested expressions which operate over
     // inputs with wide schemas. For more details on the performance issues that motivated this
     // flat, see SPARK-15680.
-    if (SparkEnv.get != null && SparkEnv.get.conf.getBoolean("spark.sql.codegen.comments", false)) {
```

Comment:
note to other reviewers: for historical context why this was done, see https://github.com/apache/spark/pull/13421/files#r65268674

Reply:
`SQLConf` is based on the active spark session. The active spark session is not always correctly set. Thus, we do not encourage the community to use `SQLConf`.

Reply:
See the PR: https://github.com/apache/spark/pull/18568

Reply:
There are places in codebase which do `SQLConf.get` (esp. `CodeGenerator.scala` which is being modified in this PR). Are you suggesting to change that everywhere like #18568 ? I think it will be good thing to do.

Reply:
Thank you for your comments. Is it better to pass `SQLConf` to the constructor of `CodegenContext`?

Reply:
I am afraid it might require a lot of code changes if we add SQLConf to `CodegenContext`.

Reply:
@tejasapatil Maybe just do nothing now. We need to fix the bugs in active session maintenance first.

Reply:
Good to hear that
@tejasapatil Could you please let us know when it is available as a PR?

Reply:
So far, I do not have a bandwidth to fix it. If anybody is interested in this, please feel free to start it. This requires a design doc at first.